core/validatorapi/grpc/validator.go

```diff
@@ -0,0 +1,512 @@
+		Target:          convert(data.Target).(*prysmpb.Checkpoint),
+	}, nil
+}
+
+// ProposeAttestation 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) ProposeAttestation(ctx context.Context, attestation *prysmpb.Attestation) (*prysmpb.AttestResponse, error) {
+	att := convert(attestation).(*eth2p0.Attestation)
+	root, _ := att.Data.HashTreeRoot()
+
+	_ = s.eth2Cl.SubmitAttestations(ctx, []*eth2p0.Attestation{att})
+
+	return &prysmpb.AttestResponse{AttestationDataRoot: root[:]}, nil
+}
+
+// SubmitAggregateSelectionProof can use integrated but probably 🆙 requires upgrade along with SubscribeCommitteeSubnets.
+func (s *BeaconNodeValidatorServer) SubmitAggregateSelectionProof(ctx context.Context, req *prysmpb.AggregateSelectionRequest) (*prysmpb.AggregateSelectionResponse, error) {
+	// TODO(corver): This is an interesting alternative to our v2 API change.
+	//  The VC provides the partial selection proof and the BN returns the complete aggregation and proof
+	//  for the VC to sign. The VC basically doesn't need to be any logic.
+
+	// Assuming we already completed `core.DutyPrepareAggregator` for this slot.
+	aggSig, _ := s.aggSigDB.Await(ctx, core.NewPrepareAggregatorDuty(int64(req.Slot)), toGroupPubkey(req.PublicKey))
+
+	// TODO(corver): We will need to add a different API to dutyDB to query the DutyAggregator data by pubkey instead of attestation hash.
+	aggAtt, _ := s.dutyDB.AwaitAggAttestation(ctx, int64(req.Slot), eth2p0.Root{} /* req.PublicKey */)
+
+	return &prysmpb.AggregateSelectionResponse{AggregateAndProof: &prysmpb.AggregateAttestationAndProof{
+		AggregatorIndex: pb.ValidatorIndex(pubkeyToValIdx(req.PublicKey)),
+		Aggregate:       convert(aggAtt).(*prysmpb.Attestation),
+		SelectionProof:  aggSig.Signature(),
+	}}, nil
+}
+
+// SubmitSignedAggregateSelectionProof 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) SubmitSignedAggregateSelectionProof(ctx context.Context, req *prysmpb.SignedAggregateSubmitRequest) (*prysmpb.SignedAggregateSubmitResponse, error) {
+	proof := convert(req).(*eth2p0.SignedAggregateAndProof)
+	_ = s.eth2Cl.SubmitAggregateAttestations(ctx, []*eth2p0.SignedAggregateAndProof{proof})
+
+	dataRoot, _ := proof.Message.Aggregate.Data.HashTreeRoot()
+
+	return &prysmpb.SignedAggregateSubmitResponse{AttestationDataRoot: dataRoot[:]}, nil
+}
+
+// ProposeExit 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) ProposeExit(ctx context.Context, exit *prysmpb.SignedVoluntaryExit) (*prysmpb.ProposeExitResponse, error) {
+	var sig eth2p0.BLSSignature
+	copy(sig[:], exit.Signature)
+
+	signedExit := &eth2p0.SignedVoluntaryExit{
+		Message:   convert(exit.Exit).(*eth2p0.VoluntaryExit),
+		Signature: eth2p0.BLSSignature{},
+	}
+
+	_ = s.eth2Cl.SubmitVoluntaryExit(ctx, signedExit)
+
+	root, _ := signedExit.Message.HashTreeRoot()
+	return &prysmpb.ProposeExitResponse{ExitRoot: root[:]}, nil
+}
+
+// SubscribeCommitteeSubnets 🆙 requires upgrade along with SubscribeCommitteeSubnets.
```

Comment:
what does this mean?

Reply:
typo, should be "similar to beaconAPI upgrade proposals"